### PR TITLE
feat: add register page

### PR DESCRIPTION
**Pull request by baseplate-admin**

### CODE REVIEW

File: backend/django_core/apps/pages/forms/user.py [hunk 1]
```diff
@@ -46,3 +46,12 @@ class RegisterForm(forms.Form):
             }
         ),
     )
+    password2 = forms.CharField(
```
Comment (baseplate-admin) [inline backend/django_core/apps/pages/forms/user.py:49]: Use confirm_password

**Review by tokitouq — APPROVED** [review]
LGTM :sparkles:

**Comment by tokitouq** [comment]
https://github.com/baseplate-admin/CoreProject/blob/5db3da20c1377b0a97ffbb730dcf0088cf3ef9fd/backend/django_core/apps/pages/views/user.py#L81-L86

add this kinda toast logic on above code as well?
show toast for each errors

```py
elif first_form.errors:
    for _, field_error in first_form.errors.as_data().items():
        for error in field_error:
            response = render(
                request,
                "components/toast.html",
                {"message": error.message},
            )
            return retarget(response, "#toast")
```

File: backend/django_core/apps/pages/views/user.py [hunk 2]
```diff
@@ -102,19 +102,21 @@ def logout_view(request: HttpRequest) -> HttpResponse:
     return redirect(redirect_location)
 
 
-def register_view(request: HttpRequest) -> HttpResponse:
+def register_view(request: HttpRequest) -> HttpResponse | None:
```
Comment (baseplate-admin) [inline backend/django_core/apps/pages/views/user.py:105]: This should never be none

**Pull request merged**